commands/pm/pm.drush.inc

```diff
@@ -2245,7 +2245,13 @@ function drush_find_empty_directories($dir, $exclude = array()) {
 function drush_pm_inject_info_file_metadata($project_dir, $project_name, $version, $datestamp) {
   // `drush_drupal_major_version()` cannot be used here because this may be running
   // outside of a Drupal context.
-  $yaml_format = substr($version, 0, 1) >= 8;
+  $yaml_format = TRUE;
+  if (
+      strpos($version, 'x') && // non-semantic versioning
```

Comment:
Semver versions might contain an x e.g. for dev branches; perhaps a regex to test for 8.x-1.* et. al. would be better.